src/main/java/org/codehaus/plexus/archiver/AbstractUnArchiver.java

```diff
@@ -373,7 +373,11 @@ else if ( isDirectory )
                 }
             }
 
-            targetFileName.setLastModified( entryDate.getTime() );
+            final long millis = entryDate.getTime();
+            if ( millis >= 0 )
+            {
+                targetFileName.setLastModified( millis );
```

Comment:
I find `setLastModified` behavior strange because `lastModified` works perfectly fine with negative values (dates before 1970-01-01). We can use `Files.setLastModifiedTime`, which works with negative values[1], and use `null` for missing date. Some archives, such as tar, support negative values so it should be up to the specific `UnArchiver` implementation to decide if the date is known or not.

[1] Whether the underlying file system supports it is different question. Unfortunately the behavior is unspecified if the date out of range for the file system. Maybe we can just catch the exception so failure to set the date does not fail the file extraction?